tsdb/mapper.go

```diff
@@ -39,7 +39,8 @@ func (mo *mapperOutput) key() string {
 // LocalMapper is for retrieving data for a query, from a given shard.
 type LocalMapper struct {
 	shard           *Shard
-	stmt            *influxql.SelectStatement
+	stmt            influxql.Statement
+	selectStmt      *influxql.SelectStatement
```

Comment:
Comment this -- "nil if the statement is not a select statement" or something similar. It's this kind of thing that makes me think there should be a separate type wrapping all this.